### ISSUE TITLE
Graph of patient still visible after being cured

### ISSUE THREAD

**Issue by Alberth289346** [CorsixTH/CorsixTH#1909]
Two patient graph issues:

1. When the patient goes home, one doesn't expect to see a declining health graph when clicked.

   I am not sure if the original does this though but it feels counter-intuitive.

2. While at it, I also noticed that the graph has 'spikes':
   ![image](https://user-images.githubusercontent.com/3254232/120105525-8dc9e100-c159-11eb-85b2-69ba12b15631.png)

   enlarged:
   ![image](https://user-images.githubusercontent.com/3254232/120105571-c23d9d00-c159-11eb-912c-272c7cc18ef8.png)

   This may be a rendering bug or it may be intentional.

**Comment by lewri** [comment]
Yes I think in original the graph would stop tracking health once cured.
Or maybe your hospital is making people sick even as they leave :)

**Comment by lewri** [comment]
The graph effectively gets frozen when the patient goes home because of `humanoid:tickDay`.
The issue is, when we cure the patient (or kill them) we update their health before we get to feed this to their health history (because of the above).

In reality, we need to process this happiness change before the graph is frozen. Then it will show the correct state. Should be relatively simply.

**Comment by ARGAMX** [comment]
0.68.0 don't see spikes on graph.

<img width="214" alt="Image" src="https://github.com/user-attachments/assets/20df44d2-9804-4d91-961b-01498e813eb9" />

original the graph stop tracking health once cured.
